doc/source/user_guide/text.rst

```diff
@@ -81,59 +86,159 @@ or convert from existing pandas data:
    s2
    type(s2[0])
 
+However there are four distinct :class:`StringDtype` variants that may be utilized.
```

Comment:
Personally, I think I would move this lower down the file, so it is not one of the first things a user reads when looking at how to use strings in pandas (not before string methods are shown for example). 
Because the file is meant as a user guide, although it probably reads more as a reference guide ..

Now, that's a bigger comment in general on the structure of the file, so let's see this comment as a possible future improvement. The PR is already a good step forward.